analyses/cell-type-wilms-tumor-06/00_fetal_reference_kidney.Rmd

```diff
@@ -0,0 +1,240 @@
+---
+title: "Build azimuth compatible fetal kidney reference from the kidney cell atlas"
+author: "Maud PLASCHKA"
+date: '2024-08-07'
+output: 
+  html_document: 
+    toc: yes
+    toc_float: yes
+    code_folding: hide
+    highlight: pygments
+    df_print: paged
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE,
+                      message=FALSE,
+                      warnings=FALSE)
+```
+
+
+Introduction
+
+The aim is to build an azimuth compatible reference for fetal kidney from the kidney cell atlas.
+The rds data can be download using the download link https://datasets.cellxgene.cziscience.com/40ebb8e4-1a25-4a33-b8ff-02d1156e4e9b.rds
+
+## Packages
+
+Load required packages in the following chunk, if needed.
+Do not install packages here; only load them with the `library()` function.
+
+```{r packages, message=FALSE, warning=FALSE}
+library("Seurat")
+library(sctransform)
+library(Azimuth)
+library(SCpubr)
+library(tidyverse)
+library(patchwork)
+library(SeuratWrappers)
+```
+
+
+## Base directories
+
+```{r base paths, eval=TRUE, include=TRUE}
+# The base path for the OpenScPCA repository, found by its (hidden) .git directory
+repository_base <- rprojroot::find_root(rprojroot::is_git_root)
+
+# The current data directory, found within the repository base directory
+data_dir <- file.path(repository_base, "data", "2024-07-08", "SCPCP000006")
+
+# The path to this module
+module_base <- file.path(repository_base, "analyses", "cell-type-wilms-tumor-06")
+```
+
+
+## Input files
+
+
+rds data can be downloaded using the URLhttps://datasets.cellxgene.cziscience.com/40ebb8e4-1a25-4a33-b8ff-02d1156e4e9b.rds
+
+Please note that this download link permanently references the current version of the dataset (08/2024). 
+If this dataset is updated, a new download link will be created that permanently references the next version of this dataset.
+
+We save it in the marker-sets folder of the module.
+Note to the DataLab: should we save it somewhere else? 
+I suggest this rds file could be placed here transiently and removed once the reference is build?
+
+
+```{r path_to_data}
+path_to_data <- file.path(module_base, "marker-sets/fetal_full.rds")
```

Comment:
For transient files, you can [put them in `scratch`](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/analysis-modules/#skeleton-analysis-module-contents).

```suggestion
We place it in the `scratch` folder of the module because it is a temporary file.


```{r path_to_data}
path_to_data <- file.path(module_base, "scratch", fetal_full.rds")
```

Files in `scratch` are ignored, which is what we want here!